test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/ODataQueryOptionTest.cs

```diff
@@ -264,6 +264,37 @@ public void CanExtractQueryOptionsCorrectly()
             Assert.Equal("DeltaToken", queryOptions.RawValues.DeltaToken);
         }
 
+
+        [Theory]
+        [InlineData(" $filter=Filter& $select=Select& $orderby=OrderBy& $expand=Expand& $top=10& $skip=20& $count=true& $skiptoken=SkipToken& $deltatoken=DeltaToken")]
+        [InlineData("%20$filter=Filter&%20$select=Select&%20$orderby=OrderBy&%20$expand=Expand&%20$top=10&%20$skip=20&%20$count=true&%20$skiptoken=SkipToken&%20$deltatoken=DeltaToken")]
+        [InlineData("$filter =Filter&$select =Select&$orderby =OrderBy&$expand =Expand&$top =10&$skip =20&$count =true&$skiptoken =SkipToken&$deltatoken =DeltaToken")]
+        public void CanExtractQueryOptionsWitLeadingSpacesCorrectly(string clause)
+        {
+            var model = new ODataModelBuilder().Add_Customer_EntityType().Add_Customers_EntitySet().GetEdmModel();
```

Comment:
> var model = new ODataModelBuilder() [](start = 11, length = 36)

would you help add // Arrange // Act // Assert

And help that in the above one?